### PR TITLE
Fix crash in server tests on windows

### PR DESCRIPTION
**Pull request by curtisman**
Loading the `node-rdkafka` module without create an instance of RdKafka would cause the node process to AV when exiting (observed in windows).  This causes our `npm run test` command to fail even though all the test passed.

The solution is to only load the module if we are using RdKafka.

### CODE REVIEW

File: server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaBase.ts [hunk 1]
```diff
@@ -26,6 +19,7 @@ export interface IKafkaEndpoints {
 }
 
 export abstract class RdkafkaBase extends EventEmitter {
+    protected readonly kafka: typeof kafkaTypes;
```
Comment (tylerbutler) [inline server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaBase.ts:22]: nit: it looks like this file uses tabs... (yuck)

File: server/routerlicious/packages/services-ordering-rdkafka/src/tryImport.ts [hunk 2]
```diff
@@ -3,14 +3,33 @@
...
+// for local dev flow. Using a pre-built image works fine (https://github.com/Blizzard/node-rdkafka/issues/315).
+// Because of this limitation, currently we cannot use node-rdkafka in local dev flow. So locally kafka config should
+// always point to kafka-node library. Production code can use either one of those.
+//
+// node-rdkafka set up an AtExit callback to call RdKafka::wait_destroyed to cleanup
+// However, on windows it AVs if we never create and rdkafka instance.
+// librdkafka lazy initialize the global mutex, and it wouldn't be initialize it if no rdkafka is created,
+// and the clean would try to access the mutex and crash.
+//
+// So we will also lazy import node-rdkafka only if we intent to use it.
```
Comment (tylerbutler) [inline server/routerlicious/packages/services-ordering-rdkafka/src/tryImport.ts:24]:
```suggestion
// So we will also lazy import node-rdkafka only if we intend to use it.
```

File: server/routerlicious/packages/services-ordering-rdkafka/src/tryImport.ts [hunk 3]
```diff
@@ -3,14 +3,33 @@
...
+// The native dependency of node-rdkafka throws an error when installing in one environment (e.g., macOS) and running
+// inside another (e.g., docker ubuntu). The issue only occurs because we volume mount code directly into docker
+// for local dev flow. Using a pre-built image works fine (https://github.com/Blizzard/node-rdkafka/issues/315).
+// Because of this limitation, currently we cannot use node-rdkafka in local dev flow. So locally kafka config should
+// always point to kafka-node library. Production code can use either one of those.
+//
+// node-rdkafka set up an AtExit callback to call RdKafka::wait_destroyed to cleanup
+// However, on windows it AVs if we never create and rdkafka instance.
+// librdkafka lazy initialize the global mutex, and it wouldn't be initialize it if no rdkafka is created,
+// and the clean would try to access the mutex and crash.
```
Comment (tylerbutler) [inline server/routerlicious/packages/services-ordering-rdkafka/src/tryImport.ts:22]:
```suggestion
// node-rdkafka sets up an AtExit callback to call RdKafka::wait_destroyed to cleanup.
// However, on windows it AVs if we never create an rdkafka instance.
// librdkafka lazy initializes the global mutex, and it wouldn't be initialized if no rdkafka is created,
// so then the cleanup callback tries to access the mutex and crashes.
```

**Review by tylerbutler — APPROVED** [review]
🏗️

File: server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaBase.ts (hunk 1 above)
Comment (curtisman) [reply server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaBase.ts:22]: converted to space

**Comment by curtisman** [comment]
@tanviraumi do you mind taking a look as well?

**Review by tanviraumi — APPROVED** [review]

**Review by GaryWilber — APPROVED** [review]

**Pull request merged**